compose/snippets/src/main/java/com/example/compose/snippets/pictureInPicture/PictureInPictureSnippets.kt

```diff
@@ -0,0 +1,384 @@
+import android.content.Intent
+import android.content.IntentFilter
+import android.graphics.drawable.Icon
+import android.os.Build
+import android.util.Rational
+import androidx.activity.ComponentActivity
+import androidx.annotation.DrawableRes
+import androidx.annotation.RequiresApi
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.toAndroidRectF
+import androidx.compose.ui.layout.boundsInWindow
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.app.PictureInPictureModeChangedInfo
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.toRect
+import androidx.core.util.Consumer
+import androidx.media3.common.Player
+import androidx.media3.exoplayer.ExoPlayer
+
+var shouldEnterPipMode by mutableStateOf(false)
+
+// [START android_compose_pip_broadcast_receiver_constants]
+// Constant for broadcast receiver
+const val ACTION_BROADCAST_CONTROL = "broadcast_control"
+
+// Intent extras for broadcast controls from Picture-in-Picture mode.
+const val EXTRA_CONTROL_TYPE = "control_type"
+const val EXTRA_CONTROL_PLAY = 1
+const val EXTRA_CONTROL_PAUSE = 2
+// [END android_compose_pip_broadcast_receiver_constants]
+
+@Composable
+fun PipListenerPreAPI12(shouldEnterPipMode: Boolean) {
+    // [START android_compose_pip_pre12_listener]
+    val currentShouldEnterPipMode = true
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O &&
+        Build.VERSION.SDK_INT < Build.VERSION_CODES.S
+    ) {
+        val context = LocalContext.current
+        DisposableEffect(context) {
+            val onUserLeaveBehavior: () -> Unit = {
+                context.findActivity()
+                    .enterPictureInPictureMode(PictureInPictureParams.Builder().build())
+            }
+            context.findActivity().addOnUserLeaveHintListener(
+                onUserLeaveBehavior
+            )
+            onDispose {
+                context.findActivity().removeOnUserLeaveHintListener(
+                    onUserLeaveBehavior
+                )
+            }
+        }
+    }
+    // [END android_compose_pip_pre12_listener]
+}
+
+@Composable
+fun VideoPlayer(
+    modifier: Modifier = Modifier
+) {
+    // [START android_compose_pip_builder_auto_enter]
+    val context = LocalContext.current
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val pipModifier = modifier.onGloballyPositioned { layoutCoordinates ->
```

Comment:
`modifier` isn't used anywhere. Is that intentional?

Reply:
This doesn't look resolved and without attaching the modifier to anything I don't think it will work